refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java

```diff
@@ -210,34 +189,32 @@ public void start(ProgressEvent event) {
 
         @Override
         public void step(ProgressEvent event) {
-            current = current + progressStep;
+            current += progressStep;
             fireProgressListenerStep((int) current);
         }
 
         @Override
         public void stop(ProgressEvent event) {
-            // do not rely on plugins;
+            /* do not rely on plugins; */
         }
+
     }
-    
+
     /**
      * do undo of previous doRefactoring()
+     *
      * @param saveAfterDone save all if true
      * @return instance of Problem or null, if everything is OK
      */
     @CheckForNull
     public Problem undoRefactoring(final boolean saveAfterDone) {
-        return Utilities.runWithOnSaveTasksDisabled(new Action<Problem>() {
-            @Override public Problem run() {
-                return reallyUndoRefactoring(saveAfterDone);
-            }
-        });
+        return Utilities.runWithOnSaveTasksDisabled(() -> reallyUndoRefactoring(saveAfterDone));
     }
-    
+
     private Problem reallyUndoRefactoring(boolean saveAfterDone) {
         try {
-            ListIterator it = internalList.listIterator(internalList.size());
-            fireProgressListenerStart(0, internalList.size()+1);
+            ListIterator internalListIterator = internalList.listIterator(internalList.size());
```

Comment:
The type of `internalListIterator` could be `ListIterator<RefactoringElementImplementation>`. This should allow the type cast a few lines down to be removed.

Reply:
It shall be done. Thank you!